dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -680,7 +680,44 @@ public Set<WorkflowVersion> requestDOIForWorkflowVersion(@ApiParam(hidden = true
         checkNotNullEntry(result);
         PublicStateManager.getInstance().handleIndexUpdate(result, StateManagerMode.UPDATE);
         return result.getWorkflowVersions();
+    }
 
+    @PUT
+    @RolesAllowed({"curator", "admin"})
+    @Timed
+    @UnitOfWork
+    @Beta
+    @Path("/{workflowId}/requestAutomaticDOI/{workflowVersionId}")
+    @Operation(operationId = "requestAutomaticDOIForWorkflowVersion", description = "Request an automatic DOI for this version of a workflow.", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME))
+    public WorkflowVersion requestAutomaticDOIForWorkflowVersion(@Parameter(hidden = true, name = "user") @Auth User user,
+        @PathParam("workflowId") Long workflowId,
+        @PathParam("workflowVersionId") Long workflowVersionId,
+        String emptyBody) {
+        Workflow workflow = workflowDAO.findById(workflowId);
+        checkNotNullEntry(workflow);
+
+        WorkflowVersion workflowVersion = workflowVersionDAO.findById(workflowVersionId);
+        checkNotNullWorkflowVersion(workflowVersion, workflow, user);
+
+        if (!ZenodoHelper.automaticallyRegisterDockstoreDOI(workflow, workflowVersion, Optional.of(user), this)) {
+            throw new CustomWebApplicationException("Could not register automatic DOI.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Or potentially just avoid the empty body. This looks a bit like a POST anyway since we're creating a child resource (DOI) under the workflow version rather than updating the workflow version

`Use POST when you want to add a child resource under resources collection.`  vs `Use PUT when we want to modify a singular resource that is already a part of resource collection. PUT replaces the resource in its entirety. Use PATCH if the request updates part of the resource.`
https://restfulapi.net/rest-put-vs-post/

Reply:
IMHO, `PATCH` is the most appropriate HTTP method.  But, given that we're trying to optimize the process for speed, let's keep it consistent with the previously-existing method and stay `PUT`.

Reply:
`PATCH` works for me too